pkgs/os-specific/darwin/cctools/apple.nix

```diff
@@ -1,118 +1,133 @@
-
-cctools = stdenv.mkDerivation rec {
-  pname = "cctools";
-  version = "973.0.1";
-
-  src = fetchurl {
-    url = "https://opensource.apple.com/tarballs/cctools/cctools-${version}.tar.gz";
-    hash = "sha256-r/6tsyyfi3R/0cLl+lN/B9ZaOaVF+Z7vJ6xj4LzSgiQ=";
-  };
-
-  patches = [
-    ./cctools-add-missing-vtool-libstuff-dep.patch
-  ];
-
-  postPatch = ''
-    for file in libstuff/writeout.c misc/libtool.c misc/lipo.c; do
-      substituteInPlace "$file" \
-        --replace '__builtin_available(macOS 10.12, *)' '0'
-    done
-    substituteInPlace libmacho/swap.c \
-      --replace '#ifndef RLD' '#if 1'
-  '';
-
-  nativeBuildInputs = [ xcbuildHook memstreamHook ];
-  buildInputs = [ libobjc llvm ];
-
-  xcbuildFlags = [
-    "MACOSX_DEPLOYMENT_TARGET=10.12"
-  ];
-
-  doCheck = true;
-  checkPhase = ''
-    runHook preCheck
-
-    Products/Release/libstuff_test
-    rm Products/Release/libstuff_test
-
-    runHook postCheck
-  '';
-
-  installPhase = ''
-    runHook preInstall
-
-    rm -rf "$out/usr"
-    mkdir -p "$out/bin"
-    find Products/Release -maxdepth 1 -type f -perm 755 -exec cp {} "$out/bin/" \;
-    cp -r include "$out/"
-
-    ln -s ./nm-classic "$out"/bin/nm
-    ln -s ./otool-classic "$out"/bin/otool
-
-    runHook postInstall
-  '';
-};
-
-ld64 = stdenv.mkDerivation rec {
-  pname = "ld64";
-  version = "609";
-
-  src = fetchurl {
-    url = "https://opensource.apple.com/tarballs/ld64/ld64-${version}.tar.gz";
-    hash = "sha256-SqQ7SqmK+uOPijzxOTqtkEu3qYmcth6H7rrQ03R1Q+4=";
-  };
-
-  postPatch = ''
-    substituteInPlace ld64.xcodeproj/project.pbxproj \
-      --replace "/bin/csh" "${tcsh}/bin/tcsh" \
-      --replace 'F9E8D4BE07FCAF2A00FD5801 /* PBXBuildRule */,' "" \
-      --replace 'F9E8D4BD07FCAF2000FD5801 /* PBXBuildRule */,' ""
-
-    sed -i src/ld/Options.cpp -e '1iconst char ldVersionString[] = "${version}";'
-  '';
-
-  nativeBuildInputs = [ xcbuildHook ];
-  buildInputs = [
-    libtapi
-    libunwind
-    llvm
-    xar
-  ];
-
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p "$out/bin"
-    find Products/Release-assert -maxdepth 1 -type f -perm 755 -exec cp {} "$out/bin/" \;
-
-    runHook postInstall
-  '';
-};
-
+{
+  fetchFromGitHub,
+  lib,
+  libobjc,
+  Libsystem,
+  libtapi,
+  libunwind,
+  llvm,
+  memstreamHook,
+  ncurses,
+  stdenv,
+  symlinkJoin,
+  tcsh,
+  xar,
+  xcbuildHook,
+}: let
+  cctools = stdenv.mkDerivation (finalAttrs: {
+    pname = "cctools";
+    version = "986";
+
+    src = fetchFromGitHub {
+      owner = "apple-oss-distributions";
+      repo = finalAttrs.pname;
+      rev = "${finalAttrs.pname}-${finalAttrs.version}";
+      hash = "sha256-Fc9Bmx6/ya8+jJuKKFcqB4gTmiV1P+SScKT6CyCXAbc=";
+    };
+
+    patches = [
+      ./cctools-add-missing-vtool-libstuff-dep.patch
+    ];
+
+    postPatch = ''
+      for file in libstuff/writeout.c misc/libtool.c misc/lipo.c; do
+        substituteInPlace "$file" \
+          --replace '__builtin_available(macOS 10.12, *)' '0'
```

Comment:
I wonder if we can drop this once the LLVM bump lands. It appears `__builtin_available` was added in Clang 12.